src/main/java/com/parasoft/findings/utils/doc/RuleDocumentationHelper.java

```diff
@@ -110,6 +110,12 @@ private String createCustomLocalLoc(URL url) {
             return docsExactMatch.getAbsolutePath();
         }
 
+        String docZipFilePath = guessRuleDocZipFile(docRoot.getAbsolutePath());
+        if (docZipFilePath != null) {
+            ZipFileUtil.setDocZipFilePath(docZipFilePath);
+            return ZipFileUtil.getRuleDocFileLocationInZip(docZipFilePath, _sRuleId);
+        }
+
         return guessRuleFile(docRoot, _sRuleId);
```

Comment:
I have a comment earlier about `guessRuleFile` method. I think it should be called before we try to access the zip file.
@rwang-parasoft And do you know what this logic is for? It looks like it's looking for a file using rule id as name as a fallback file, but I don't see any file like that in the package. Could it be that older versions of the language tools used this structure?

Reply:
I believe this method was intended to allow finding the rule documentation even when the user specified an incorrect rule doc root path. However, I now think this method is unnecessary. If the user specifies the wrong rule doc root path, the rule documentation should not be found, and the user should be required to correct the location of the rule documentation.
In the latest implementation, this method has already been removed.